jcl/src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -413,12 +413,18 @@ MethodType accessModeType(Class<?> receiver, Class<?> type, Class<?>... args) {
 		}
 
 		this.fieldType = setter.parameterType(setter.parameterCount() - 1);
-		this.coordinateTypes = getter.parameterArray();
+		
+		/* The first VarHandle parameter type should be removed from the getter in order to derive
+		 * the coordinate types. 
+		 */
+		this.coordinateTypes = getter.dropFirstParameterType().parameterArray();
```

Comment:
Rather than creating a new MT here (`getter.dropFirstParameterType`), can you just copy a subset of the originalMT's parameterArray?

Something like
```suggestion
		Class<?>[] getterTypes = getter.parameterArray();
		this.coordinateTypes = Arrays.copyOfRange(getterTypes, 1, getterTypes.length);
```

This still has one additional allocation (the getterTypes array) but saves the more costly MT creation

Reply:
Updated.